ocs_ci/deployment/mce.py

```diff
@@ -0,0 +1,243 @@
+"""
+This module contains functionality required for mce installation.
+"""
+
+import logging
+import tempfile
+
+from ocs_ci.framework import config
+from ocs_ci.ocs.resources.ocs import OCS
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.utility import templating
+from ocs_ci.ocs import constants
+from ocs_ci.utility.utils import (
+    run_cmd,
+    exec_cmd,
+)
+from ocs_ci.ocs import exceptions
+from ocs_ci.ocs.resources.catalog_source import CatalogSource
+from ocs_ci.ocs import ocp
+from ocs_ci.utility.utils import get_running_ocp_version
+
+logger = logging.getLogger(__name__)
+
+
+class MCEInstaller(object):
+    """
+    mce Installer class for mce deployment
+    """
+
+    def __init__(self):
+        self.namespace = constants.MCE_NAMESPACE
+        self.ns_obj = ocp.OCP(kind=constants.NAMESPACES)
+        self.hypershift_override_image_cm = "hypershift-override-images-new"
+        self.multicluster_engine = ocp.OCP(
+            kind="MultiClusterEngine",
+            resource_name=constants.MULTICLUSTER_ENGINE,
+        )
+
+    def create_mce_catalog_source(self):
+        """
+        Creates a catalogsource for mce operator.
+
+        """
+        logger.info("Adding CatalogSource for MCE")
+        mce_catalog_source_data = templating.load_yaml(constants.MCE_CATSRC_YAML)
+        mce_catalog_source_name = mce_catalog_source_data.get("metadata").get("name")
+        if config.ENV_DATA.get("mce_image"):
+            mce_image_tag = config.ENV_DATA.get("mce_image")
+        mce_catalog_source_data["spec"]["image"] = mce_image_tag
+        mce_catalog_source_manifest = tempfile.NamedTemporaryFile(
+            mode="w+", prefix="mce_catalog_source_manifest", delete=False
+        )
+        templating.dump_data_to_temp_yaml(
+            mce_catalog_source_data, mce_catalog_source_manifest.name
+        )
+        run_cmd(f"oc apply -f {mce_catalog_source_manifest.name}", timeout=2400)
+        mce_catalog_source = CatalogSource(
+            resource_name=mce_catalog_source_name,
+            namespace=constants.MARKETPLACE_NAMESPACE,
+        )
+
+        # Wait for catalog source is ready
+        mce_catalog_source.wait_for_state("READY")
+
+    def create_mce_namespace(self):
+        """
+        Creates the namespace for mce resources
+
+        Raises:
+            CommandFailed: If the 'oc create' command fails.
+        """
+        try:
```

Comment:
it is better to check if resource exist than using try-except, to not pollute logs with false-errors